btr-api/src/btr_api/resources/submission.py

```diff
@@ -40,20 +40,134 @@
+        'birthDate': REDACT_DATE,
+        'identifiers': REDACT_IDENTIFIER,
+    },
+}
+
+
+def redactInformation(request, payload):
+    role = btr_auth.getUserType(request)
+    if role == btr_auth.USER_COMPETENT_AUTHORITY:
+        return payload
+    # otherwise PUBLIC or STAFF
+    redactionToUse = REDACT_RULES[role]
+    for person in payload['payload']['personStatements']:
+        for name in person['names']:
+            if name['type'] == 'alternative':
+                name['fullName'] = redactField(name['fullName'], redactionToUse.get('prefName'))
+
+        person['email'] = redactField(person['email'], redactionToUse.get('email'))
+        person['phoneNumber']['number'] = redactField(person['phoneNumber']['number'], redactionToUse.get('phone'))
+        for address in person['addresses']:
+            address['postalCode'] = redactField(address['postalCode'], redactionToUse.get('postal'))
+            address['street'] = redactField(address['street'], redactionToUse.get('street'))
+            address['streetAdditional'] = redactField(
+                address['streetAdditional'], redactionToUse.get('streetAdditional')
+            )
+            address['locationDescription'] = redactField(
+                address['locationDescription'], redactionToUse.get('locationDescription')
+            )
+
+        person['placeOfResidence']['postalCode'] = redactField(
+            person['placeOfResidence']['postalCode'], redactionToUse.get('postal')
+        )
+        person['placeOfResidence']['street'] = redactField(
+            person['placeOfResidence']['street'], redactionToUse.get('street')
+        )
+        person['placeOfResidence']['streetAdditional'] = redactField(
+            person['placeOfResidence']['streetAdditional'], redactionToUse.get('streetAdditional')
+        )
+        person['placeOfResidence']['locationDescription'] = redactField(
+            person['placeOfResidence']['locationDescription'], redactionToUse.get('locationDescription')
+        )
+        person['birthDate'] = redactField(person['birthDate'], redactionToUse.get('birthDate'))
+        for identifier in person['identifiers']:
+            identifier['id'] = redactField(identifier['id'], redactionToUse.get('identifiers'))
+    return jsonify(payload)
+
+
+def redactField(field, redactType):
+    if redactType == REDACT_MONONYM:
+        return field[0:1] + '***'
+    elif redactType == REDACT_MONONYM_FN:
+        rv = ''
+        for word in field.split():
+            if not (rv == ''):
+                rv += ' '
+            rv += word[0:1] + '***'
+        return rv
+    elif redactType == REDACT_EMAIL:
+        return field.split('@')[0][0:1] + '***' + '@***.' + field.split('.')[-1]
+    elif redactType == REDACT_PHONE:
+        return field[0:-7] + '***'
+    elif redactType == REDACT_FULL:
+        return '***'
+    # This is a space instead of blank because if it's empty the ui shows undefined in some spots
```

Comment:
keep it as is, but maybe we need to discuss this. As it might get messy if we keep putting spaces in the values.